Unknown CMake command "python3_add_library"

Hi, I am not sure if the issue belong here or is a pybind's downstream issue, but [point to the line 406](https://github.com/OPM/opm-common/blob/release/2023.04/rc1/CMakeLists.txt#L406) and I am leaving this record.

```bash
$ cmake -S opm-common-release-2023.04-rc1 -B build-cmake
==> Retrieving sources...
  -> Downloading opm-common-release-2023.04-rc1.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 3877k    0 3877k    0     0  3326k      0 --:--:--  0:00:01 --:--:-- 3873k
100 9948k    0 9948k    0     0  4595k      0 --:--:--  0:00:02 --:--:-- 4974k
100 13.3M    0 13.3M    0     0  4310k      0 --:--:--  0:00:03 --:--:-- 4544k
100 14.9M    0 14.9M    0     0  4351k      0 --:--:--  0:00:03 --:--:-- 4565k
==> Validating source files with sha512sums...
    opm-common-release-2023.04-rc1.tar.gz ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting opm-common-release-2023.04-rc1.tar.gz with bsdtar
==> Starting build()...
-- The C compiler identification is GNU 12.2.1
-- The CXX compiler identification is GNU 12.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/sbin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/sbin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at cmake/Modules/Findopm-tests.cmake:26 (message):
  opm-tests not found - integration tests using opm-tests will be skipped.
Call Stack (most recent call first):
  CMakeLists.txt:62 (include)


-- Performing Test HAVE_C99
-- Performing Test HAVE_C99 - Success
-- Found C99: -std=c99  
-- Could NOT find Valgrind (missing: VALGRIND_INCLUDE_DIR VALGRIND_PROGRAM) 
-- Found Boost: /usr/lib/cmake/Boost-1.81.0/BoostConfig.cmake (found suitable version "1.81.0", minimum required is "1.44.0") found components: system unit_test_framework 
-- Performing Test HAVE_CJSON
-- Performing Test HAVE_CJSON - Success
-- Found cjson: /usr/include/cjson  
-- Performing Test QuadMath_COMPILES
-- Performing Test QuadMath_COMPILES - Success
-- Found QuadMath: (Supported by compiler)  
-- Found Python3: /usr/sbin/python3.10 (found version "3.10.10") found components: Interpreter 
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- Found pybind11: /usr/include (found version "2.10.4")
-- Performing Test HAVE_ATTRIBUTE_ALWAYS_INLINE
-- Performing Test HAVE_ATTRIBUTE_ALWAYS_INLINE - Success
-- Performing Test HAS_ATTRIBUTE_UNUSED
-- Performing Test HAS_ATTRIBUTE_UNUSED - Success
-- Performing Test HAS_ATTRIBUTE_DEPRECATED
-- Performing Test HAS_ATTRIBUTE_DEPRECATED - Success
-- Performing Test HAS_ATTRIBUTE_DEPRECATED_MSG
-- Performing Test HAS_ATTRIBUTE_DEPRECATED_MSG - Success
-- CMake version: 3.26.3
-- Linux distribution: Arch Linux
-- Target architecture: x86_64
-- Found Git: /usr/sbin/git (found version "2.40.0") 
-- Source code repository: not found!
-- GNU C++ compiler version: 12.2.1 20230201
-- Linker: ld 2.40
Processing opm_defaults opm-common
-- Checking to see if CXX compiler accepts flag -Wl,--enable-new-dtags
-- Checking to see if CXX compiler accepts flag -Wl,--enable-new-dtags - yes
-- Precompiled headers: disabled
-- Build type: None
-- Checking to see if CXX compiler accepts flag -mtune=native
-- Checking to see if CXX compiler accepts flag -mtune=native - yes
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Checking to see if CXX compiler accepts flag -pthread
-- Checking to see if CXX compiler accepts flag -pthread - yes
-- Could NOT find CppCheck (missing: CPPCHECK_PROGRAM) 
-- Disabling clang-check as CMAKE_EXPORT_COMPILE_COMMANDS is not enabled
-- Could NOT find Valgrind (missing: VALGRIND_INCLUDE_DIR VALGRIND_PROGRAM) 
-- Generating debug symbols: -ggdb3
-- Looking for strip utility
-- Looking for strip utility - found
-- Performing Test HAVE_DYNAMIC_BOOST_TEST
-- Performing Test HAVE_DYNAMIC_BOOST_TEST - Success
-- Writing config file "/tmp/makepkg/opm-common/src/build-cmake/config.h"...
-- This build defaults to installing in /usr
-- Checking whether the file system is case-sensitive
-- File system is case-sensitive
-- Found Doxygen: /usr/sbin/doxygen (found version "1.9.6") found components: doxygen dot 
-- Writing version information to local header project-version.h
-- Looking for sgemm_
-- Looking for sgemm_ - not found
-- Looking for sgemm_
-- Looking for sgemm_ - found
-- Found BLAS: /usr/lib/libblas.so  
-- Looking for cheev_
-- Looking for cheev_ - not found
-- Looking for cheev_
-- Looking for cheev_ - found
-- Found LAPACK: /usr/lib/liblapack.so;/usr/lib/libblas.so  
-- Performing Test HAVE_CXA_DEMANGLE
-- Performing Test HAVE_CXA_DEMANGLE - Success
-- Found MPI_C: /usr/lib/libmpi.so (found version "3.1") 
-- Found MPI_CXX: /usr/lib/libmpi_cxx.so (found version "3.1") 
-- Found MPI: TRUE (found version "3.1")  
-- Found GMP: /usr/lib/libgmpxx.so  
-- Found dune-common: /usr/include  
-- Looking for MPI_Finalized
-- Looking for MPI_Finalized - found
-- Version 2.9.0 of dune-common from /usr/lib/cmake/dune-common
CMake Error at /usr/lib/cmake/pybind11/pybind11NewTools.cmake:189 (python3_add_library):
  Unknown CMake command "python3_add_library".
Call Stack (most recent call first):
  CMakeLists.txt:406 (pybind11_add_module)


-- Configuring incomplete, errors occurred!
==> ERROR: A failure occurred in build().
    Aborting...
==> Removing installed dependencies...
checking dependencies...
:: gettext optionally requires git: for autopoint infrastructure updates
:: glib2 optionally requires python: gdbus-codegen, glib-genmarshal, glib-mkenums, gtester-report
:: glibc optionally requires gd: for memusagestat
:: groff optionally requires netpbm: for use together with man -H command interaction in browsers
:: groff optionally requires libxaw: for gxditview
:: iproute2 optionally requires python: for routel
:: libevent optionally requires python: event_rpcgen.py
:: libxml2 optionally requires python: Python bindings
:: systemd optionally requires python: Unified Kernel Image with ukify
:: util-linux optionally requires python: python bindings to libmount
warning: dependency cycle detected:
warning: harfbuzz will be removed after its freetype2 dependency
```

- Full log attached [here](https://github.com/OPM/opm-common/files/11226619/job.txt).

#### Relevant information

- pybind11 version: 2.10.4
- opm-common version: 2023.04 rc1
- dune-common version: 2.9.0
- cmake version: 3.26.3
- os: arch linux

#### Related links

- https://discourse.cmake.org/t/unknown-cmake-command-python3-add-library/7799
- https://github.com/pybind/pybind11/issues/3996

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -216,15 +216,15 @@ if (OPM_ENABLE_PYTHON)
     if(PYTHON_EXECUTABLE AND NOT Python3_EXECUTABLE)
       set(Python3_EXECUTABLE ${PYTHON_EXECUTABLE})
     endif()
+    # We always need to search for Development as we use
+    # pybind11_add_module even if don't embed Python
+    find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
     if (OPM_ENABLE_EMBEDDED_PYTHON)
-      find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
       get_target_property(_lib_path Python3::Python IMPORTED_LOCATION)
       set(PYTHON_LIBRARY ${_lib_path})
       set(PYTHON_LIBRARIES {PYTHON_LIBRARY})
       list(APPEND opm-common_LIBRARIES ${PYTHON_LIBRARY})
       set(PYTHON_INCLUDE_DIRS ${Python3_INCLUDE_DIRS})
-    else()
-      find_package(Python3 REQUIRED COMPONENTS Interpreter)
     endif()
     # Make sure we fail gracefully here without setuptool
     execute_process(COMMAND ${Python3_EXECUTABLE} -c "import setuptools"

